### ISSUE TITLE
Some albums are not displayed in search results

### ISSUE THREAD

**Issue by Zbuben** [jaedb/spotmop#112]
Hi,

I've got an issue searching some artists. Whereas songs are shown in the search results, some albums show until you specifically search them or browse them directly from tracks.

To reproduce, search for "Mathieu Boogaerts", in the overview page the "super 2" album is missing but in the main search result page you can see some of its song (ondule for example).

When I use another client (e.g. moped or native spotify client) this album is present.
When I look for this particular album, I find it with its title and I can go back on artist's overview clicking the artist name below the album title... 

Regards,

Christophe.

ps: I'm using spotmop 2.7.3 installed with pip.

**Comment by jaedb** [comment]
Issue identified as being due to batching script to fetch additional album data. Basically the results were < 20 so the batch script wigged out. Sorted now and will be bundled in the next release.

Feel free to test the pre-release on the `develop` branch.

**Comment by Zbuben** [comment]
Wow, that's efficient ;)
Thanks !

edit: tested here, it works like a charm.
edit2: tested a bit more, in the search results it works, but not in the overview of the artist, I get the same truncated album list than before.

**Comment by Zbuben** [comment]
Hi,

you perfectly fixed the search tool, but the artist view is still missing some albums. Should I open another issue or this one will make the trick ?

Regards,

Christophe.

**Comment by Zbuben** [comment]
Gasp, I figured out why this artist had missing albums in his overview, it's a problem with the market zone declared in the spotify database and is totally upstream, nothing to blame on the spotmop side. 

The albums displayed are indeed those given by spotify api when the market is used in the URL, no one is missing. I tried to remove this information and I got all albums I was expecting. 

To perform a full test, I just removed the ?market=country suffix in the url used by getArtistAlbums in  spotmop/src/app/services/spotify/service.js (line 1361 in the develop branch)... I don't know if it's a good idea to get rid of it, actually.

This issue has to remain closed ;)

Regards,

Christophe.
